### PR TITLE
internal: add type to mutateData

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/angular/generator.ts [hunk 1]
```diff
@@ -202,7 +202,7 @@ export default class AngularGenerator extends BaseApplicationGenerator {
 
             return returnValue;
           },
-        });
+        } as any);
```
Comment (Tcharl) [inline generators/angular/generator.ts:205]: Why isn't it as `F`?

File: generators/app/support/config.ts [hunk 2]
```diff
@@ -152,7 +153,7 @@ export const loadDerivedAppConfig = ({ application }: { application: any }) => {
 
     projectDescription: ({ projectDescription, baseName }) => projectDescription ?? `Description for ${baseName}`,
     endpointPrefix: ({ applicationType, lowercaseBaseName }) => (applicationType === 'microservice' ? `services/${lowercaseBaseName}` : ''),
-  });
+  } as any);
```
Comment (Tcharl) [inline generators/app/support/config.ts:156]: as ApplicationType?

File: generators/base-application/support/prepare-entity.ts [hunk 3]
```diff
@@ -285,7 +289,7 @@ export function derivedPrimaryKeyProperties(primaryKey) {
     typeLong: primaryKey.type === LONG,
     typeInteger: primaryKey.type === INTEGER,
     typeNumeric: !primaryKey.composite && primaryKey.fields[0].fieldTypeNumeric,
-  });
+  } as any);
```
Comment (Tcharl) [inline generators/base-application/support/prepare-entity.ts:292]: as PK?

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
@Tcharl sorry, did not see your reviews before I committed

**Comment by mshima** [comment]
@Tcharl `mutateData` now expects a Record as context but is receiving any objects, the others parameters expects a type based on context type.
Those data doesn't match current application/entity/field/relationship types and are too many changes to handle at once.
That's the reason that as any are required.

**Comment by Tcharl** [comment]
@DanielFran @mshima No problem! I even think I've made the review post merge.
I'm trying to understand the best practice on the core gen to contribute properly and inline with the conventions.
I'm also sorry to bother you with stupid questions, typescript is not my native programming language and these ones help me to provide better contribution. Clean core is a something that is important and I want to help as much as I can

**Pull request merged**